crates/rust-analyzer/src/lsp_ext.rs

```diff
@@ -337,3 +337,31 @@ pub struct CommandLink {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub tooltip: Option<String>,
 }
+
+pub enum OpenDocs {}
+
+impl Request for OpenDocs {
+    type Params = OpenDocsParams;
+    type Result = DocumentationLink;
+    const METHOD: &'static str = "rust-analyzer/openDocs";
+}
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OpenDocsParams {
+    // TODO: I don't know the difference between these two methods of passing position.
+    #[serde(flatten)]
+    pub position: lsp_types::TextDocumentPositionParams,
+    // pub textDocument: lsp_types::TextDocumentIdentifier,
+    // pub position: lsp_types::Position,
+}
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DocumentationLink {
+    pub remote: String, // TODO: Better API?
+                        // #[serde(skip_serializing_if = "Option::is_none")]
+                        // pub remote: Option<String>,
+                        // #[serde(skip_serializing_if = "Option::is_none")]
+                        // pub local: Option<String>
+}
```

Comment:
Don't forget to document the new extension in https://github.com/rust-analyzer/rust-analyzer/blob/753af410056937d7be3a1b3457c834a887288957/docs/dev/lsp-extensions.md :-)

Reply:
Noticed a few send (should be sent) typos in there, is that intended to match a method name or something?

Reply:
No, it's just me mixing up English tenses :)